src/etching.ts

```diff
@@ -1,10 +1,22 @@
+import { None, Option, Some } from '@sniptt/monads';
```

Comment:
why not just using undefined? This package doesn't seem to be very popular on npm.

Reply:
This is emulating rust's monad structure, it was easier to use this to ensure code is equivalent to casey's.